### ISSUE TITLE
zh-Hant and zh-Hans do not resolve to zh-CN and zh-HK

### ISSUE THREAD

**Issue by wldevries** [Humanizr/Humanizer#626]
We use the mainland China and Hong Kong cultures to indicate the languages. Unfortunately Humanizer only supports zh-Hant and zh-Hans and treats zh-HK and zh-CN as if they are English. They should be mapped to the traditional and simplified cultures.

**Comment by MaStr11** [comment]
Currently humanizer has three resource entries for Chinese. 

* zh-CN
* zh-Hans
* zh-Hant

zh-CN and zh-Hans are completely identical. 

The output for different cultures is:
```
zh-CN   : 6 个月, 17 天
zh-Hans : 6 个月, 17 天
zh-Hant : 6 個月, 17 天
zh-HK   : 6 months, 17 days
zh-TW   : 6 months, 17 days
zh      : 6 months, 17 days
en-US   : 6 months, 17 days
```

We could change the resource definition from zh-CN to zh and delete zh-Hans. This would result in the following output:

```
zh-CN   : 6 个月, 17 天
zh-Hans : 6 个月, 17 天
zh-Hant : 6 個月, 17 天
zh-HK   : 6 个月, 17 天
zh-TW   : 6 个月, 17 天
zh      : 6 个月, 17 天
en-US   : 6 months, 17 days
```

All existing tests would pass after that change. This would change the current fall-back logic:

Before:

* *if culture is zh-CN, zh-Hans or zh-Hant*: Use the corresponding resource files for the translation.
* *Otherwise*: Fall-back to standard culture (en-US?).

After:

* *if culuture is zh-Hant*: Use zh-Hant.
* *if culture is zh or zh-??*: Use zh.

Further readings:
[Resource fall-back process](https://docs.microsoft.com/en-us/dotnet/articles/framework/resources/packaging-and-deploying-resources-in-desktop-apps?view=netframework-4.7)
[Chinese culture codes](https://msdn.microsoft.com/en-us/library/cc233982.aspx)

If the new logic is the desired behavior I can create an PR. @wldevries can you confirm the proposed change.

**Comment by wldevries** [comment]
I'm not sure if that'll work. According to [documentation](https://msdn.microsoft.com/en-us/library/dd997383(v=vs.100).aspx) the parent chain for zh-CN is as follows:

> zh-CN → zh-CHS → zh-Hans → zh → Invariant

However for some reason the zh-Hans code is not used as a fallback. This may be caused by manually matching cultures in the Humanizr code base, I'm not sure.

**Comment by MaStr11** [comment]
Thank you for pointing that out. It seems that the current code base do not use the standard fall-back mechanism. The current implementation uses this approach:

* Register known cultures (about 50 in total) in [FormatterRegistry.cs](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/Configuration/FormatterRegistry.cs)
* If the requested culture is registered return this "localizer" [here](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/Configuration/LocaliserRegistry.cs#L72)
* Otherwise check whether the TwoLetter culture is registered and use it as fall-back [here](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/Configuration/LocaliserRegistry.cs#L75)
* Otherwise take the default localizer [here](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/Configuration/LocaliserRegistry.cs#L78)

These questions remain:

1. Humanizer only supports two variations of Chinese (Hant and Hans). Humanizer publishes nuget packages for zh-CN, ch-Hans (both with identical funtionality) and zh-Hant. To better reflect what is really supported by Humanizer we should publish packages for 
   * zh (as common basis for ch-CN, zh-Hans and all other zh cultures e.g zh-HK or zh-TW) and 
   * zh-Hant.
2. The logic in [LocaliserRegistry.FindLocaliser](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/Configuration/LocaliserRegistry.cs#L72) should be changed to use CultureInfo.Parent as fall-back mechanism because CultureInfo.Parent for zh-CN [returns the correct fall-back sequence](https://dotnetfiddle.net/2m5dvE).

We need some feedback by the owners (@MehdiK, @onovotny, @hazzik, @mexx) because this might result in a change of the published [nuget packages](https://github.com/Humanizr/Humanizer/blob/dev/NuSpecs/Humanizer.nuspec#L53).

**Comment by hazzik** [comment]
Implemented option 2) in #638
